File: Question_02.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Function to count the number of words in a line
function countWords(line) {
  return line.trim().split(/\s+/).length;
}

rl.question('Enter the value of N: ', (N) => {
  N = parseInt(N); // Convert N to an integer
  const lines = [];
  let linesReceived = 0; // Keep track of the lines received

  // Read lines of input
  rl.on('line', (line) => {
    if (linesReceived < N) {
      const wordCount = countWords(line);

      if (wordCount <= 100) {
        lines.push(line);
        linesReceived++;

        // Print the received line
        console.log(`Words Input in Line ${linesReceived}: ${line}`);

        if (linesReceived === N) {
          rl.close();
        }
      } else {
        console.log("Number of words exceeded the limit. Closing application.");
        rl.close();
      }
    }
  });

  // After all input is received, process and display the reversed words
  rl.on('close', () => {
    lines.forEach((line) => {
      const words = line.split(' ');
      const reversedWords = words.map((word) => {
        return word.split('').reverse().join('');
      });
      console.log(reversedWords.join(' '));
    });
  });
});

File: question_02.py
# Define a function to reverse the order of characters in each word of a line
def reverse_words(line):
    # Split the line into individual words
    words = line.split()

    # Create a list of reversed words using list comprehension
    reversed_words = [word[::-1] for word in words]

    # Join the reversed words to form a line with reversed words
    return " ".join(reversed_words)


# Read the number of lines
n = int(input(""))

# Read all lines of input and store them in a list
lines = []
for _ in range(n):
    # Read a line of input from the user and append it to the 'lines' list
    line = input()
    lines.append(line)

# Process and output reversed lines
for line in lines:
    # Call the 'reverse_words' function to reverse words in the current line
    reversed_line = reverse_words(line)

    # Print the reversed line
    print(reversed_line)
